src/v/pandaproxy/kafka_client_cache.cc

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright 2022 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#include "pandaproxy/kafka_client_cache.h"
+
+#include "pandaproxy/client_cache_error.h"
+#include "pandaproxy/logger.h"
+#include "utils/gate_guard.h"
+
+#include <chrono>
+
+using namespace std::chrono_literals;
+
+namespace pandaproxy {
+
+static constexpr auto clean_timer_period = 10s;
+
+kafka_client_cache::kafka_client_cache(
+  YAML::Node const& cfg,
+  size_t max_size,
+  std::vector<config::broker_authn_endpoint> kafka_api,
+  model::timestamp::type keep_alive)
+  : _config{cfg}
+  , _cache_max_size{max_size}
+  , _kafka_has_sasl{false}
+  , _keep_alive{keep_alive} {
+    // Is there a Kafka listener with SASL enabled?
+    auto ep_it = std::find_if(
+      kafka_api.begin(),
+      kafka_api.end(),
+      [](const config::broker_authn_endpoint& ep) {
+          return ep.authn_method == config::broker_authn_method::sasl;
+      });
+
+    _kafka_has_sasl = ep_it != kafka_api.end();
```

Comment:
I don't think `enable_authorization` matters here because I have not seen any routes on PP where we need to use the authorization API. So ACls are enforced on the Kafka API side.

Reply:
I shall remove because it's not for this project.